lib/chef_client_run.go

```diff
@@ -93,7 +103,8 @@ func ChefClientRun(config *Config, nodeName string, firstRun bool, requests chan
 	}
 
 	if config.RunChefClient {
-		if firstRun {
+
+		if firstRun && !skipClientCreation {
```

Comment:
Verify that we also want to skip node load/creation if `skipClientCreation` is true